Update_and_Start_Bot.bat

```diff
@@ -9,10 +15,18 @@ curl https://api.github.com/repos/%Owner%/%Repo%/releases/latest > response.txt
 FOR /F "tokens=*" %%g IN ('FIND "tag_name" "response.txt"') do set result=%%g
 set "tag_name=%result:"tag_name": "=%"
 set "tag_name=%tag_name:",=%"
-
-curl -L -o %filename% https://github.com/%Owner%/%Repo%/archive/%tag_name%/%filename%
-tar -zxvf %filename% -C "%CD%" --strip-components=1
-del /f %filename%
 del /f response.txt
 
-py -3 main.py
+>nul FIND "Current_Version" ".env" || (echo: >> ".env" | echo Current_Version = 0 >> ".env")
+FOR /f "tokens=1,2 delims== " %%G in ('FIND "Current_Version" ".env"') do set current_version=%%H
+
+IF NOT %current_version%==%tag_name% (
+    findstr /v /i /L /c:"Current_Version" ".env" > ".tmp"
+    move /Y ".tmp" ".env" > nul
+    echo Current_Version = %tag_name% >> ".env"
+    curl -L -o %filename% https://github.com/%Owner%/%Repo%/archive/%tag_name%/%filename%
+    tar -zxvf %filename% -C "%CD%" --strip-components=1
+    del /f %filename%
+) else (
+    py -3 main.py
+)
```

Comment:
This still doesn't do what we need it to. Think about the user here, or even the script triggering this script. 

When this is run, you expect the bot to update and run. You can't expect the user (or the triggering script) to need to check if the bot is running after this script is ran every time. However, at the same time, you can't run the script off the same instance that you've updated the contents of the folder with. There might be a change in launch parameters for example.

Reply:
Regardless, this is going into the backlog seeing that we'll probably move into a unix based system for this.